server/pkg/nlslayer/data_time.go

```diff
@@ -0,0 +1,25 @@
+package nlslayer
+
+type Time struct {
+	Property          *string
+	Interval          *int
+	UpdateClockOnLoad *bool
+}
+
+func (t *Time) SetProperty(property *string) {
+	t.Property = property
+}
+
+func (t *Time) SetInterval(interval *int) {
+	t.Interval = interval
+}
+
+func (t *Time) SetUpdateClockOnLoad(updateClockOnLoad *bool) {
+	t.UpdateClockOnLoad = updateClockOnLoad
+}
+
+func (t *Time) Update(property *string, interval *int, updateClockOnLoad *bool) {
+	t.Property = property
+	t.Interval = interval
+	t.UpdateClockOnLoad = updateClockOnLoad
+}
```

Comment:
what if `t` is nil?

Reply:
Are you suggest that I should add construction function like
```
func NewTime() *Time {
   return &Time{}
}
```

or I should add the checks at the beginning of each method? like
```
func (t *Time) SetProperty(property *string) {
   if t == nil { return }
   t.Property = property
}
```